test/Microsoft.ML.Tests/DatabaseLoaderTests.cs

```diff
@@ -79,6 +79,63 @@ public void IrisLightGbm()
             }).PredictedLabel);
         }
 
+        [LightGBMFact]
+        public void IrisLightGbmConnectionTimeout()
```

Comment:
How is this testing the timeout?

Reply:
what would you recommend that the test look like?

---
In reply to: [349252966](https://github.com/dotnet/machinelearning/pull/4494#discussion_r349252966) [](ancestors = 349252966)

Reply:
I'm not exactly sure. But my initial thinking is doing something that would make the command timeout, and ensuring the correct behavior occurs.

Reply:
added test where SQL query has delay added, and command timeout is exercised.

---
In reply to: [349721564](https://github.com/dotnet/machinelearning/pull/4494#discussion_r349721564) [](ancestors = 349721564)

Reply:
Should you also test for user initiated cancellation? I wonder if this way of testing introduces test failures on the infra or problems with other databases than SQL Server?

As for an example, here is one another way: https://github.com/dotnet/orleans/blob/585680f3c06cb82f22e07b65330b769d333a8aae/test/Extensions/TesterAdoNet/StorageTests/RelationalStoreTests.cs#L179. (For context, this is actually testing timeouts while streaming data. This piece isn't the best possible otherwise, but should give perhaps food for thoughts in the future.)

Reply:
> Should you also test for user initiated cancellation? I wonder if this way of testing introduces test failures on the infra or problems with other databases than SQL Server?
> 
> As for an example, here is one another way: https://github.com/dotnet/orleans/blob/585680f3c06cb82f22e07b65330b769d333a8aae/test/Extensions/TesterAdoNet/StorageTests/RelationalStoreTests.cs#L179. (For context, this is actually testing timeouts while streaming data. This piece isn't the best possible otherwise, but should give perhaps food for thoughts in the future.)

Could you elaborate on what 'user-initiated cancellation' would mean in this context?

Reply:
@ashbhandare I was thinking that why fix the command timeout at https://github.com/dotnet/machinelearning/pull/4494/files#diff-6bd895b22ba4c7e741ba7526ac214e5aR33 when it could be per query? While maybe also providing an overload to a user supplied cancellation token and then writing a test that excercises that token and timeout. But maybe I misunderstood this and `DatabaseSource` is intended to use per query or this is the behaviour hoped.

Sorry for the delay, I lost the plot for a while.